src/Resources/Adjustments/Operations/CreateAdjustment.php

```diff
@@ -22,21 +22,39 @@ class CreateAdjustment implements \JsonSerializable
      */
     public function __construct(
         public readonly Action $action,
-        public readonly array|Undefined $items,
+        public readonly array|Undefined|null $items,
         public readonly string $reason,
         public readonly string $transactionId,
         public readonly AdjustmentType|Undefined $type = new Undefined(),
     ) {
-        if ($this->type === AdjustmentType::Partial() && ($this->items instanceof Undefined || empty($this->items))) {
+        $typeIsFull = AdjustmentType::Full()->equals($this->type);
+
+        if (! $typeIsFull && ($this->items instanceof Undefined || empty($this->items))) {
             throw InvalidArgumentException::arrayIsEmpty('items');
         }
+
+        if ($typeIsFull && is_array($this->items)) {
+            throw new InvalidArgumentException('items are not allowed when the adjustment type is full');
```

Comment:
API returns error _"Array must have at least 1 items"_ when an empty array is sent in request, and _"items are not allowed when the adjustment type is full"_ when items are provided with `full` type